### PR TITLE
return Nyaplot plot instance instead of nil. 

### PR DESCRIPTION
**Pull request by Shekharrajak**
This will help me to use the Nyaplot plot object for further configuration like weight, height, etc in daru-view : https://github.com/Shekharrajak/daru-view/pull/3 .

IRuby notebook will still display plot.

### CODE REVIEW

**Comment by Shekharrajak** [comment]
Ping @zverok @v0dro @lokeshh

File: lib/daru/plotting/nyaplot/category.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def plot opts
             # Set yrange for good view
             set_yrange plot, opts[:method]
             yield plot, diagram if block_given?
-            plot.show
```
Comment (Shekharrajak) [inline lib/daru/plotting/nyaplot/category.rb]: `def plot` returns nothing previously.

File: spec/plotting/nyaplot/vector_spec.rb [hunk 2]
```diff
@@ -37,7 +34,6 @@
     context type.to_s do
       it 'does not set x axis' do
         expect(plot).to receive(:add).with(type, [11, 22, 33]).ordered
-        expect(plot).to receive(:show).ordered
 
         vector.plot(type: type)
```
Comment (Shekharrajak) [inline spec/plotting/nyaplot/vector_spec.rb]: Should I add line 

```
        expect(vector).to receive(:plot).with(type: type).and_return(plot)
        vector.plot(type: type)
```
? Is it correct way to do this ? 
Ping @zverok

Comment (Shekharrajak) [inline spec/plotting/nyaplot/vector_spec.rb]: Is there any other good way for `InstanceDouble` ?

Comment (zverok) [inline spec/plotting/nyaplot/vector_spec.rb]: I believe this should be enough:
```ruby
expect(vector.plot(type: type)).to eq plot
```
It exactly checks that `#plot` method would return an instance of plot constructed, or am I missing something?

Comment (Shekharrajak) [inline spec/plotting/nyaplot/vector_spec.rb]: Fine! Previously I was comparing the class using `be_a`, `eq` , instance of class. Also didn't remove the `vector.plot(type: type)` line, that's why I was getting some `InstanceDouble ` error. Now I have updated the testcases.
Thanks!

**Comment by v0dro** [comment]
@Shekharrajak can you confirm that you've checked that this works in iRuby? I'm good with this change if yes.

**Comment by Shekharrajak** [comment]
@v0dro ,Yes I have checked it. It is working fine.

**Comment by v0dro** [comment]
Ok. Let @zverok approve and we can merge.

**Comment by Shekharrajak** [comment]
Ping @v0dro

**Pull request merged**